File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int front = 0 , rear = nums.size()-1;
        while(rear >= front){
            int middle = (rear + front) / 2;
            if (nums[middle] == target){
                return middle;
            }else if(nums[middle] < target){
                    front = middle + 1;
            }else{
                    rear = middle - 1;
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        front,rear = 0,len(numbers)-1
        while(rear>front):
            if numbers[front] + numbers[rear] > target:
                rear-=1
            elif numbers[front] + numbers[rear] < target:
                front+=1
            elif numbers[front] + numbers[rear] == target:
                ans=[]
                ans.append(front+1)
                ans.append(rear+1)
                return ans